clients/ops/admin-ui/src/features/datastore-connections/add-connection/manual/DSRCustomizationForm.tsx

```diff
@@ -0,0 +1,163 @@
+import { Box, HStack, Text, VStack } from "@fidesui/react";
+import { useAlert } from "common/hooks";
+import { TrashCanSolidIcon } from "common/Icon";
+import { FieldArray, Form, Formik } from "formik";
+import { useRouter } from "next/router";
+import React from "react";
+import * as Yup from "yup";
+
+import { STEPS } from "../constants";
+import CustomInput from "../forms/CustomInput";
+import { ButtonGroup as ManualButtonGroup } from "./ButtonGroup";
+import { Field } from "./types";
+
+type DSRCustomizationFormProps = {
+  data: Field[];
+  isSubmitting: boolean;
+  onSaveClick: (values: any, actions: any) => void;
+};
+
+const DSRCustomizationForm: React.FC<DSRCustomizationFormProps> = ({
+  data = [],
+  isSubmitting = false,
+  onSaveClick,
+}) => {
+  const router = useRouter();
+  const { errorAlert } = useAlert();
+
+  const handleCancel = () => {
+    router.push(STEPS[1].href);
+  };
+
+  const handleSubmit = (values: any, actions: any) => {
+    const uniqueValues = new Set(values.fields.map((f: Field) => f.pii_field));
+    if (uniqueValues.size < values.fields.length) {
+      errorAlert("PII Field must be unique");
+      return;
+    }
+    onSaveClick(values, actions);
+  };
+
+  return (
+    <Formik
+      initialValues={{
+        fields:
+          data.length > 0
+            ? data
+            : ([
+                {
+                  pii_field: "",
+                  dsr_package_label: "",
+                },
+              ] as Field[]),
+      }}
+      onSubmit={handleSubmit}
+      validateOnBlur={false}
+      validateOnChange={false}
+      validationSchema={Yup.object({
+        fields: Yup.array().of(
+          Yup.object().shape({
+            pii_field: Yup.string().required("PII Field is required"),
+            dsr_package_label: Yup.string().required(
+              "DSR Package Label is required"
```

Comment:
This is fine, just noting that it is not required from a backend perspective. If the DSR Package Label doesn't exist, I create it from the PII Field by lowercasing it and adding underscores in between. Often I imagine they might want them to just be the same too.